### PR TITLE
Upgrade to pgx 0.5.4

### PR DESCRIPTION
**Pull request by JamesGuthrie**
## Description

Waiting on PR to be merged upstream [1].

[1]: https://github.com/tcdi/pgx/pull/716

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [ ] CHANGELOG entry for user-facing changes
- [ ] Updated the relevant documentation

### CODE REVIEW

File: src/raw.rs [hunk 1]
```diff
@@ -57,6 +58,15 @@ macro_rules! raw_type {
 }
 
 #[derive(Clone, Copy)]
-pub struct bytea(pub pg_sys::Datum);
+pub struct bytea(pub Datum);
+
+unsafe impl<'a> SqlTranslatable for bytea {
```
Comment (workingjubilee) [inline src/raw.rs:63]:
```suggestion
#[allow(clippy::extra-unused-lifetimes] // pgx sorcery caused this
unsafe impl<'a> SqlTranslatable for bytea {
```

File: src/raw.rs [hunk 2]
```diff
@@ -57,6 +58,15 @@ macro_rules! raw_type {
...
+    fn argument_sql() -> Result<SqlMapping, ArgumentError> {
+        Ok(SqlMapping::literal("bytea"))
+    }
+    fn return_sql() -> Result<Returns, ReturnsError> {
+        Ok(Returns::One(SqlMapping::literal("bytea")))
```
Comment (workingjubilee) [inline src/raw.rs:68]: I'm surprised. I thought we had a mapping for `bytea`.

File: src/aggregates/gapfill_delta.rs [hunk 3]
```diff
@@ -34,22 +34,22 @@ pub mod _prom_ext {
...
         is_counter: bool,
         is_rate: bool,
     }
 
     impl GapfillDeltaTransition {
         pub fn new(
-            lowest_time: pg_sys::TimestampTz,
-            greatest_time: pg_sys::TimestampTz,
+            lowest_time: i64,
+            greatest_time: i64,
```
Comment (workingjubilee) [inline src/aggregates/gapfill_delta.rs:52]:
```suggestion
            lowest_time: TimestampWithTimeZone,
            greatest_time: TimestampWithTimeZone,
```

File: src/aggregates/gapfill_delta.rs [hunk 4]
```diff
@@ -72,7 +72,7 @@ pub mod _prom_ext {
             }
         }
 
-        pub(crate) fn add_data_point(&mut self, time: pg_sys::TimestampTz, val: f64) {
+        pub(crate) fn add_data_point(&mut self, time: i64, val: f64) {
```
Comment (workingjubilee) [inline src/aggregates/gapfill_delta.rs:75]:
```suggestion
        pub(crate) fn add_data_point(&mut self, time: TimestampWithTimeZone, val: f64) {
```

File: src/aggregates/gapfill_delta.rs [hunk 5]
```diff
@@ -90,7 +90,7 @@ pub mod _prom_ext {
             }
         }
 
-        fn in_current_window(&self, time: pg_sys::TimestampTz) -> bool {
+        fn in_current_window(&self, time: i64) -> bool {
```
Comment (workingjubilee) [inline src/aggregates/gapfill_delta.rs:93]:
```suggestion
        fn in_current_window(&self, time: TimestampWithTimeZone) -> bool {
```

**Review by workingjubilee — COMMENTED** [review]
Not entirely sure all cases should go over to TimestampWithTimeZone, actually, but I'll let you be the judge.

**Comment by sumerman** [comment]
> Not entirely sure all cases should go over to TimestampWithTimeZone, actually, but I'll let you be the judge.

Thank you @workingjubilee! AFAICT they all should.

**Comment by sumerman** [comment]
> > Not entirely sure all cases should go over to TimestampWithTimeZone, actually, but I'll let you be the judge.
> 
> Thank you @workingjubilee! AFAICT they all should.

Unfortunately, I'll have to revert this. Even though, semantically, it seems like something we'd want, I couldn't find a way to perform arithmetic ops on `TimestampWithTimeZone`: its internal field is private, and it doesn't implement `Add`

**Review by jgpruitt — APPROVED** [review]

**Pull request merged**